docs/src/main/asciidoc/se/injection.adoc

```diff
@@ -0,0 +1,850 @@
+
+[source,java]
+.Custom annotation for interception
+----
+include::{sourcedir}/se/inject/InterceptorExample.java[tag=snippet_1, indent=0]
+----
+
+=== Interceptor interface
+The link:{service-registry-base-url}/io/helidon/service/registry/Interception.Interceptor.html[`Interception.Interceptor`]
+interface defines an interceptor service that intercepts methods/constructors/fields annotated with the configured marker annotation.
+The interceptor service must be named by the fully qualified name of the
+link:{service-registry-base-url}/io/helidon/service/registry/Interception.Intercepted.html[`Interception.Intercepted`]
+annotation (either via
+link:{service-registry-base-url}/io/helidon/service/registry/Service.Named.html[`@Service.Named`] or
+link:{service-registry-base-url}/io/helidon/service/registry/Service.NamedByType.html[`@Service.NamedByType`]).
+
+To properly handle the interception chain, the interceptor must always invoke the `proceed` method
+if the invocation should continue normally.
+However, it is also possible to return a custom value directly from the interceptor and
+effectively bypass the original method execution.
+
+[source,java]
+.Sample Interceptor interface implementation
+----
+include::{sourcedir}/se/inject/InterceptorExample.java[tag=snippet_2, indent=0]
+----
+
+<1> Binds this Interceptor to process elements annotated with `@Traced`
+<2> Passing interceptor processing to another interceptor in the chain
+
+=== Delegate annotation
+The link:{service-registry-base-url}/io/helidon/service/registry/Interception.Delegate.html[`@Interception.Delegate`]
+annotation enables interception for classes that aren’t created through the service registry
+but are instead produced by a factory (More about factories can be found here - <<Factories, Factory chapter>>).
+
+Let's make the same `@Traced` annotation and Interceptor as in the previous examples
+
+[source,java]
+.Custom annotation and interceptor
+----
+include::{sourcedir}/se/inject/InterceptorDelegateExample.java[tag=snippet_1, indent=0]
+----
+
+Now, let's create the factory of the service instance.
+
+[source,java]
+.Instance producer
+----
+include::{sourcedir}/se/inject/InterceptorDelegateExample.java[tag=snippet_2, indent=0]
+----
+
+Method calls on an instance created this way can’t be intercepted. To enable interception in such cases, we use the
+link:{service-registry-base-url}/io/helidon/service/registry/Interception.Delegate.html[`@Interception.Delegate`] annotation.
+However, keep in mind that usage of this annotation doesn’t add the ability to intercept constructor calls.
+To enable interception, this annotation must be present on the class that the factory produces.
+While it is not required on interfaces, it will still work correctly if applied there.
+
+If you need to enable interception for classes using delegation, you should make sure about the following:
+
+- The class must have accessible no-arg constructor (at least package local) and mustn’t be final
```

Comment:
`mustn't` seems wrong to me here, maybe some native speaker can help

Reply:
And it is still missing the requirement that the class itself has to be extensible (i.e. not `final`), as we are generating a subclass to handle interception.

Reply:
Hm... that was supposed to be that `mustn’t be final`. But perhaps it looked like I am talking about the constructor?